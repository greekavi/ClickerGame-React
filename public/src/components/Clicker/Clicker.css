h2{
    
    font-family: Century Gothic,CenturyGothic,AppleGothic,sans-serif; 
}
p{
    font-family: Century Gothic,CenturyGothic,AppleGothic,sans-serif; 
    font-size: 20px;
}
#button1{
    background-color: #62055f;
    font-size: 30px;
    color: white;
}
#button1:hover{
    background-color:#f688f7 ;
    color:   #62055f;
}
