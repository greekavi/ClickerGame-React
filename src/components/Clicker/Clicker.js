import React,{useState} from "react";
import Button from '@material-ui/core/Button';
import './Clicker.css';
function Clicker({ClickerCount,ClickerTick}){  
    return(
<div>
<p>You Clicked <b>{ClickerCount}</b> times</p>
          
                          
            <button id="button1"  onClick={ClickerTick}>
            Start Clicking!
      </button>
      </div>
    );
}
export default Clicker;