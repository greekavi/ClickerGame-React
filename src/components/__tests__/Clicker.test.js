import React from 'react';
import ReactDom from 'react-dom';
import {render,fireEvent,cleanup,screen,container} from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import "@testing-library/jest-dom/extend-expect";
import {shallow,configure} from 'enzyme';
import Clicker from './../Clicker/Clicker';
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() })

afterEach(() => {
    jest.clearAllMocks();
  });

  it("Check if Clicking button works",()=>{
    const ClickerTick=jest.fn();
    const wrapper=shallow(<button id="button1"  onClick={ClickerTick}>Start Clicking!</button>);
    wrapper.find("#button1").simulate("click");
    expect(ClickerTick.mock.calls.length).toEqual(1);
  })

  it("Passes variables correctly from counter to Clicker",()=>{
    const {getByTestId}=render(<Clicker ClickerCount="10"/>);
    expect(getByTestId('clickerCountId')).toHaveTextContent("10");
})

it("Matches Snapshot",()=>{
  const tree=renderer.create(<Clicker TimerSeconds="10"/>).toJSON();
  expect(tree).toMatchSnapshot();
})