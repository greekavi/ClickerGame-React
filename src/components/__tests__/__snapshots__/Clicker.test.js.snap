// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div>
  <p
    data-testid="clickerCountId"
  >
    You Clicked 
    <b />
     times
  </p>
  <button
    id="button1"
  >
    Start Clicking!
  </button>
</div>
`;
